### PR TITLE
Update Cub Cache Creation to new Method

### PR DESCRIPTION
**Pull request by tylera-nvidia**
update all the code in cub.cu to use the new method of creating cub plans for the cache

### CODE REVIEW

File: include/matx/transforms/cub.h
Comment (cliffburdick) [inline include/matx/transforms/cub.h:1233]: What is this for?

File: include/matx/transforms/cub.h
Comment (cliffburdick) [inline include/matx/transforms/cub.h:1279]: Same

**Review by cliffburdick — CHANGES_REQUESTED** [review]

File: include/matx/transforms/cub.h
Comment (tylera-nvidia) [reply include/matx/transforms/cub.h:1279]: I have another comment in this PR explaining why I needed this and asking for a second opinion on how to get around it. I'm not sure how to link to another comment in the same PR... it's at line [1276](https://github.com/NVIDIA/MatX/pull/694/files/1969207f7a78218fe53226b6d3cc0f211d68e2be#diff-7c667b17610a2198f407d07c43ae3ed0aa7f9d9777c0bd7d084df613b5d9d441)

Comment (tylera-nvidia) [reply include/matx/transforms/cub.h:1279]: (I see now they were pending and I forgot to submit them).

File: include/matx/transforms/cub.h
Comment (tylera-nvidia) [reply include/matx/transforms/cub.h]: updated to remove int and use correct empty param based on our discussion

File: include/matx/transforms/cub.h
Comment (tylera-nvidia) [reply include/matx/transforms/cub.h]: updated to remove int and use correct empty param based on our discussion

Comment (tylera-nvidia) [reply include/matx/transforms/cub.h]: updated to remove int and use correct empty param based on our discussion

**Comment by tylera-nvidia** [comment]
@cliffburdick  I've updated the plan creations per our discussion yesterday. can you review/approve this PR when you get a chance?

**Review by cliffburdick — APPROVED** [review]

**Pull request merged**
